### PR TITLE
add test for #6836

### PR DESCRIPTION
**Pull request by snosov1**
@vpisarev, @vicproon I've converted the reproducing sample from #6836 to a regression test

### CODE REVIEW

File: modules/calib3d/test/test_cameracalibration.cpp [hunk 1]
```diff
@@ -1346,6 +1346,108 @@ void CV_ProjectPointsTest_CPP::project( const Mat& objectPoints, const Mat& rvec
 
 ///////////////////////////////// Stereo Calibration /////////////////////////////////////
 
+class CV_StereoCalibrationCornerTest : public cvtest::BaseTest
+{
```
Comment (ilya-lavrenov) [inline modules/calib3d/test/test_cameracalibration.cpp]: please, avoid obsolete `cvtest::BaseTest`-based tests. Use pure Google Test tests.

Comment (snosov1) [inline modules/calib3d/test/test_cameracalibration.cpp]: I follow the "consistency is better than brilliance" principle. Every other test in this file looks like this. So, I follow the common pattern. If the conversion should happen, it should, probably, happen all at once.

Comment (ilya-lavrenov) [inline modules/calib3d/test/test_cameracalibration.cpp]: :-1:

Comment (snosov1) [inline modules/calib3d/test/test_cameracalibration.cpp]: Ok, we'll need an arbiter =) @vpisarev what do you think?

**Comment by ilya-lavrenov** [comment]
@snosov1, please, fix http://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/6288/steps/compile%20release/logs/warnings%20%281%29

File: modules/calib3d/test/test_cameracalibration.cpp [hunk 2]
```diff
@@ -1346,6 +1346,108 @@ void CV_ProjectPointsTest_CPP::project( const Mat& objectPoints, const Mat& rvec
...
+    image(cv::Rect(imageSize.width / 3, imageSize.height / 3, imageSize.width / 3, imageSize.height / 3)).setTo(255);
+
+    // perform remap-resize
+    Mat src_result;
+    remap(image, src_result, srcRmap[0], srcRmap[1], INTER_LINEAR);
+    resize(src_result, src_result, Size(), scale, scale, INTER_LINEAR);
+    // perform resize-remap
+    Mat rsz_result;
+    resize(image, rsz_result, Size(), scale, scale, INTER_LINEAR);
+    remap(rsz_result, rsz_result, rszRmap[0], rszRmap[1], INTER_LINEAR);
```
Comment (vpisarev) [inline modules/calib3d/test/test_cameracalibration.cpp]: this part looks a bit weird to me. You check that resize(remap(img)) == remap(resize(img)), which is generally not true, because of rounding operations and many other reasons; It can only be true if img is constant image. In your test the image is simple but non-constant, and this is strange that it passed the test, probably, the results should be visualized to see what's going on

Comment (snosov1) [inline modules/calib3d/test/test_cameracalibration.cpp]: This is what happens - we work on an image with a constant rectangle area close the the center (moved in top-left direction). Prior to change (buggy behavior) - it would completely disappear from one of the images. After the change (correct behavior) - it stays roughly where it was. You are correct in that the borders of the rectangle do not match perfectly (because of rounding, etc.) - but below I check only the inner part (3.1 divisor instead of 3) to work around this.

I agree, that it's not the tidiest test ever. But still, it seems kind of ok to me as a "regression" test for a corner-case issue.

**Comment by vpisarev** [comment]
I see. Ok, let's merge it in! :+1:
